actors/builtin/multisig/multisig_test.go

```diff
@@ -311,6 +310,359 @@ func TestApprove(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	multisigWalletAdd := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+	darlene := newIDAddr(t, 104)
+
+	const noUnlockDuration = int64(0)
+	const numApprovals = int64(1)
+	var initialSigner = []addr.Address{anne, bob}
+
+	builder := mock.NewBuilder(context.Background(), t, multisigWalletAdd).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	testCases := []swapTestCase{
+		{
+			desc:   "happy path signer swap",
+			to:     chuck,
+			from:   bob,
+			expect: []addr.Address{anne, chuck},
+			code:   exitcode.Ok,
+		},
+		{
+			desc:   "fail to swap when from signer not found",
+			to:     chuck,
+			from:   darlene,
+			expect: []addr.Address{anne, chuck},
+			code:   exitcode.ErrPlaceholder,
+		},
+		{
+			desc:   "fail to swap when to signer already present",
+			to:     bob,
+			from:   chuck,
+			expect: []addr.Address{anne, chuck},
+			code:   exitcode.ErrPlaceholder,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			rt := builder.Build()
+
+			actor.constructAndVerify(rt, numApprovals, noUnlockDuration, initialSigner...)
+
+			rt.SetCaller(multisigWalletAdd, builtin.AccountActorCodeID)
+			rt.ExpectValidateCallerAddr(multisigWalletAdd)
+			if tc.code != exitcode.Ok {
+				rt.ExpectAbort(tc.code, func() {
+					actor.swapSigners(rt, tc.from, tc.to)
+				})
+			} else {
+				actor.swapSigners(rt, tc.from, tc.to)
+				var st multisig.MultiSigActorState
+				rt.Readonly(&st)
+				assert.Equal(t, tc.expect, st.Signers)
+			}
+			rt.Verify()
+		})
+	}
+}
+
+type thresholdTestCase struct {
+	desc             string
+	initialThreshold int64
+	setThreshold     int64
+	code             exitcode.ExitCode
+}
+
+func TestChangeThreshold(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	multisigWalletAdd := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+
+	const noUnlockDuration = int64(0)
+	var initialSigner = []addr.Address{anne, bob, chuck}
+
+	testCases := []thresholdTestCase{
+		{
+			desc:             "happy path decrease threshold",
+			initialThreshold: 2,
+			setThreshold:     1,
+			code:             exitcode.Ok,
+		},
+		{
+			desc:             "happy path simple increase threshold",
+			initialThreshold: 2,
+			setThreshold:     3,
+			code:             exitcode.Ok,
+		},
+		{
+			desc:             "fail to set threshold to zero",
+			initialThreshold: 2,
+			setThreshold:     0,
+			code:             exitcode.ErrPlaceholder,
+		},
+		{
+			desc:             "fail to set threshold less than zero",
+			initialThreshold: 2,
+			setThreshold:     -1,
+			code:             exitcode.ErrPlaceholder,
+		},
+		{
+			desc:             "fail to set threshold above number of signers",
+			initialThreshold: 2,
+			setThreshold:     int64(len(initialSigner) + 1),
+			code:             exitcode.ErrPlaceholder,
+		},
```

Comment:
Please file an issue for "what happens when threshold is reduced below the number of approvers an existing transaction already has"